src/xcode/ENA/ENA/Source/Workers/SQLiteKeyValueStore.swift

```diff
@@ -163,3 +191,55 @@ class SQLiteKeyValueStore {
 		}
 	}
 }
+
+/// Keychain Extension for storing and loading the Database Key in the Keychain
+extension SQLiteKeyValueStore {
+	func savetoKeychain(key: String, data: Data) -> OSStatus {
+		let query = [
+			kSecClass as String: kSecClassGenericPassword as String,
+			kSecAttrAccount as String: key,
+			kSecValueData as String: data ] as [String: Any]
+
+		SecItemDelete(query as CFDictionary)
+		return SecItemAdd(query as CFDictionary, nil)
+	}
+
+	func loadFromKeychain(key: String) -> Data? {
+		let query = [
+			kSecClass as String: kSecClassGenericPassword,
+			kSecAttrAccount as String: key,
+			kSecReturnData as String: kCFBooleanTrue!,
+			kSecMatchLimit as String: kSecMatchLimitOne] as [String: Any]
+
+		var dataTypeRef: AnyObject?
+		let status: OSStatus = SecItemCopyMatching(query as CFDictionary, &dataTypeRef)
+		if status == noErr {
+			return dataTypeRef as? Data ?? nil
+		} else {
+			return nil
+		}
+	}
+
+	func generateDatabaseKey() -> String? {
+		var bytes = [UInt8](repeating: 0, count: 32)
+		let result = SecRandomCopyBytes(kSecRandomDefault, bytes.count, &bytes)
+		guard result == errSecSuccess else {
+			logError(message: "Error creating random bytes.")
+			return nil
+		}
+		return "x'"+Data(bytes).hexEncodedString()+"'"
+	}
+}
+
+/// Extensions for Hexencoding when generating key
+extension Data {
+	struct HexEncodingOptions: OptionSet {
+		let rawValue: Int
+		static let upperCase = HexEncodingOptions(rawValue: 1 << 0)
+	}
+
+	func hexEncodedString(options: HexEncodingOptions = []) -> String {
```

Comment:
May I suggest to simply remove `HexEncodingOptions` completely? As far I can tell `hexEncodedString` is only ever called without any options. This is just code that is not used by anyone. Or do you have some use case in mind?

Reply:
That is true. It is not going to be needed, so removed with https://github.com/corona-warn-app/cwa-app-ios/pull/85/commits/511354c0e70698651c96f1835de360bb9c654504